src/runtime/uv/event_loop.cpp

```diff
@@ -99,43 +100,95 @@ void event_loop_run_loop(event_loop_t * event_loop) {
 
         uv_mutex_unlock(&event_loop->mutex);
     }
+
+    uv_mutex_lock(&event_loop->mutex);
+    event_loop->thread_alive = false;
+
+    uv_cond_signal(&event_loop->cond_var);
+    uv_mutex_unlock(&event_loop->mutex);
+}
+
+void event_loop_destroy(event_loop_t * event_loop) {
```

Comment:
I am surprised that you are not calling `uv_stop` before `uv_loop_close`. Also, I think (?) that the callback in `uv_walk` should actually call `uv_close` on the handles?